src/Storage/Device/S3.php

```diff
@@ -657,11 +659,17 @@ public function getPartitionTotalSpace(): float
      * Get all files and directories inside a directory.
      *
      * @param  string  $dir Directory to scan
-     * @return string[]
+     * @return array
+     *
+     * @throws Exception
      */
     public function getFiles(string $dir): array
     {
-        throw new Exception('Not implemented.');
+        $data = $this->listObjects($dir);
+        $data['IsTruncated'] = $data['IsTruncated'] === 'true';
+        $data['KeyCount'] = intval($data['KeyCount']);
+
+        return $data;
```

Comment:
Let's add a `?string $cursor = null` param to`getFiles` function, and pass that through to `listObjects` as the continuation token. Then if needed, developers could continue with the cursor. It will be irrelevant for local since we don't paginate there